Intersections_2/include/CGAL/Intersections_2/Segment_2_Segment_2.h

```diff
@@ -215,8 +242,8 @@ do_intersect_with_info(const typename K::Segment_2 &seg1,
     typename K::Compare_xy_2 compare_xy;
 
   // first try to filter using the bbox of the segments
-    if (less_xy(A2,B1)
-     || less_xy(B2,A1))
+    if (certainly(less_xy(A2,B1))
```

Comment:
@afabri this change is not related to the patch but I fixed it while I was in this file. Do you agree that it does not make sense to have a filtered failure in that case?

Reply:
Latest the `switch(make_certain(..))` will lead to the exception being thown, so we can throw it here as well.